empress/support_files/js/tree-controller.js

```diff
@@ -0,0 +1,475 @@
+     *
+     *
+     * @param{Number} i The index corresponding to a node in the ORIGINAL tree
+     *
+     * @return{String}
+     */
+    TreeController.prototype.name = function (i) {
+        return this.model.fullTree.name(i);
+    };
+
+    /**
+     * Returns an array of all node names in sheared tree.
+     */
+    TreeController.prototype.getAllNames = function () {
+        return this.model.shearedTree.getAllNames();
+    };
+
+    /**
+     * Returns the number of leaf nodes in sheared tree
+     *
+     * @return {Number}
+     */
+    TreeController.prototype.numleaves = function () {
+        return this.model.shearedTree.numleaves();
+    };
+
+    /**
+     * Returns the length of the ith index in the ORIGINAL bp tree.
+     *
+     * Note: The input of this method should the result of either preorderselect
+     *       or postorderselect.
+     *
+     * @param{Number} i The index corresponding to a node in the ORIGINAL tree
+     *
+     * @return{Number}
+     */
+    TreeController.prototype.length = function (i) {
+        return this.model.fullTree.length(i);
+    };
+
+    /**
+     * Return the parent index of the node that corresponds to the ith index in
+     * the ORIGINAL bp tree.
+     *
+     * Note: The input of this method should the result of either preorderselect
+     *       or postorderselect.
+     *
+     * Note: The output of this method is also in relation to the original tree.
+     *
+     * @param{Number} i The index corresponding to a node in the ORIGINAL tree
+     *
+     * @return{Number}
+     */
+    TreeController.prototype.parent = function (i) {
+        return this.model.fullTree.parent(i);
+    };
+
+    /**
+     * Returns the index of the opening index of the root node.
+     *
+     * Note: This will always be 0.
+     *
+     * @return {Number}
+     */
+    TreeController.prototype.root = function () {
+        return this.model.fullTree.root();
+    };
+
+    /**
+     * Returns true if i represents a leaf node
+     *
+     * Note: The input of this method should the result of either preorderselect
+     *       or postorderselect.
+     *
+     * @param{Number} i The index corresponding to a node in the ORIGINAL tree
+     *
+     * @return {Boolean}
+     */
+    TreeController.prototype.isleaf = function (i) {
+        return this.model.fullTree.isleaf(i);
+    };
+
+    /**
+     * This method is used in fchild, lchild, nsibling, and psibling and it what
+     * allows TreeController to use the topology of the sheared tree but return
+     * the results w.r.t the original tree.
+     *
+     * @param{Number} i The index correspond to a node in the ORIGINAL tree.
+     * @param{String} func The function to use. This should only be fchild,
+     *                     nchild, nsibling or psibling.
+     *
+     * @return{Number} The result of func w.r.t the ORIGINAL tree.
+     */
+
+    TreeController.prototype._curToOrigNodeFunction = function (i, func) {
```

Comment:
I think this is the last method that needs to be renamed here and in the codebase.

```suggestion
    TreeController.prototype._shearedToFullNodeFunction = function (i, func) {
```